duckduckgo/Android
# ViewBinding migration examples

Task/Issue URL: https://app.asana.com/0/488551667048375/1200358286721665/f
Tech Design URL: 
CC: 

**Description**:
This PR introduces ViewBinding in our codebase and shows how to migrate from kotlin synthetics for different scenarios.

Activities:

- AboutDuckduckGoActivity
- FeedbackActivity

Fragments
- All Feedback related Fragments (InitialFeedbackFragment, ShareOpenEndedFeedbackFragment, etc.)

Dialogs
- DaxDialog

RecyclerView.ViewHolder
- AppIconsAdapter
- SuggestionViewHolderFactory: SearchSuggestionViewHolder, BookmarkSuggestionViewHolder,

CustomView
- TabSwitcherButton



**Steps to test this PR**:
Perform a smoke tests around the components we have migrated to ViewBinding:

1. Fresh install
2. Go through the onboarding, ensure Dax dialogs appear as usual
1. Ensure About DuckDuckgo screen works as expected
2. Ensure Feedback flow (inside settings) works as expected
3. Create a few tasks and ensure Tab button is updated and works as expected
4. Ensure autocomplete works as expected


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
